lib/client/config.go

```diff
@@ -0,0 +1,343 @@
+	"critical": LogLevelCritical,
+}
+var bgpLevelToLog = map[LogLevel]string{
+	LogLevelNone:     "none",
+	LogLevelDebug:    "debug",
+	LogLevelInfo:     "info",
+	LogLevelWarning:  "none",
+	LogLevelError:    "none",
+	LogLevelCritical: "none",
+}
+
+// ConfigInterface provides methods for setting, unsetting and retrieving low
+// level config options.
+type ConfigInterface interface {
+	SetNodeToNodeMesh(bool) error
+	GetNodeToNodeMesh() (bool, error)
+	SetDefaultASNumber(uint64) error
+	GetDefaultASNumber() (uint64, error)
+	SetDefaultIPIP(bool) error
+	GetDefaultIPIP() (bool, error)
+	SetNodeIPIPTunnelAddress(string, *net.IP) error
+	GetNodeIPIPTunnelAddress(string) (*net.IP, error)
+	SetDefaultLogLevel(LogLevel) error
+	GetDefaultLogLevel() (LogLevel, error)
+	SetNodeLogLevel(string, LogLevel) error
+	SetNodeLogLevelUseDefault(string) error
+	GetNodeLogLevel(string) (LogLevel, ConfigLocation, error)
+}
+
+// config implements ConfigInterface
+type config struct {
+	c *Client
+}
+
+// newConfig returns a new ConfigInterface bound to the supplied client.
+func newConfigs(c *Client) ConfigInterface {
+	return &config{c}
+}
+
+// The configuration interface provides the ability to set and get low-level,
+// or system-wide configuration options.
+
+// SetNodeToNodeMesh sets the enabled state of the system-wide node-to-node mesh.
+// When this is enabled, each calico/node instance automatically establishes a
+// full BGP peering mesh between all nodes that support BGP.
+func (c *config) SetNodeToNodeMesh(enabled bool) error {
+	b, _ := json.Marshal(nodeToNodeMesh{Enabled: enabled})
+	_, err := c.c.backend.Apply(&model.KVPair{
+		Key:   model.GlobalBGPConfigKey{Name: "node_mesh"},
+		Value: string(b),
+	})
+	return err
+}
+
+// GetNodeToNodeMesh returns the current enabled state of the system-wide
+// node-to-node mesh option.  See SetNodeToNodeMesh for details.
+func (c *config) GetNodeToNodeMesh() (bool, error) {
+	var n nodeToNodeMesh
+	if s, err := c.getValue(model.GlobalBGPConfigKey{Name: "node_mesh"}); err != nil {
+		return false, err
+	} else if s == nil {
+		return true, nil
+	} else if err = json.Unmarshal([]byte(*s), &n); err != nil {
+		return false, err
+	} else {
+		return n.Enabled, nil
+	}
+}
+
+// SetDefaultASNumber sets the default AS Number used by the BGP agent running
+// on each node.  This may be overridden by an explicitly configured value in
+// the node resource.
+func (c *config) SetDefaultASNumber(asNumber uint64) error {
+	_, err := c.c.backend.Apply(&model.KVPair{
+		Key:   model.GlobalBGPConfigKey{Name: "as_num"},
+		Value: strconv.FormatUint(asNumber, 10),
+	})
+	return err
+}
+
+// SetDefaultASNumber gets the default AS Number used by the BGP agent running
+// on each node.  See SetDefaultASNumber for more details.
+func (c *config) GetDefaultASNumber() (uint64, error) {
+	if s, err := c.getValue(model.GlobalBGPConfigKey{Name: "as_num"}); err != nil {
+		return 0, err
+	} else if s == nil {
+		return 64511, nil
```

Comment:
Is now a good time to fix this default?  64511 is in the wrong block IIRC.

It should at least be one const, I think.